core/domain/exp_domain_test.py

```diff
@@ -1334,11 +1334,7 @@ def test_validation(self):
 
         # Restore a valid exploration.
         self.set_interaction_for_state(init_state, 'TextInput')
-        answer_groups_list = [
-            state_domain.AnswerGroup.from_dict(answer_group)
-            for answer_group in [answer_groups]
-            ]
-        init_state.update_interaction_answer_groups(answer_groups_list)
+        init_state.update_interaction_answer_groups(answer_groups)
```

Comment:
Yes, this was confusing, but there was also a lot of bad naming in some list comprehensions and I assume it might have interfered with the value, and it probably worked in a different way in py2 and py3. I have done some other changes to this test case.

Reply:
Looks good now. Thanks!